tests/integration/test_sqs.py

```diff
@@ -1404,6 +1406,110 @@ def test_batch_send_with_invalid_char_should_succeed(self, sqs_client, sqs_creat
         result_send = sqs_client.send_message_batch(QueueUrl=queue_url, Entries=batch)
         assert len(result_send["Failed"]) == 1
 
+    # TODO: Why are certain attributes "unsupported"
+    def test_get_specific_queue_attribute_response(self, sqs_client, sqs_create_queue):
+        queue_name = "queue-{}".format(short_uid())
+        dead_letter_queue_name = "dead_letter_queue-{}".format(short_uid())
+
+        dl_queue_url = sqs_create_queue(QueueName=dead_letter_queue_name)
+        dl_result = sqs_client.get_queue_attributes(
+            QueueUrl=dl_queue_url, AttributeNames=["QueueArn"]
+        )
+
+        assert "QueueArn" in dl_result["Attributes"].keys()
+        dl_queue_arn = dl_result["Attributes"]["QueueArn"]
+
+        _redrive_policy = {
+            "deadLetterTargetArn": dl_queue_arn,
+            "maxReceiveCount": "10",
+        }
+        attributes = {
+            "MessageRetentionPeriod": "604800",
+            "DelaySeconds": "10",
+            "RedrivePolicy": json.dumps(_redrive_policy),
+        }
+
+        queue_url = sqs_create_queue(QueueName=queue_name, Attributes=attributes)
+        unsupported_attributes = sqs_client.get_queue_attributes(
+            QueueUrl=queue_url,
+            AttributeNames=["MessageRetentionPeriod", "RedrivePolicy"],
+        )
+        supported_attributes = sqs_client.get_queue_attributes(
+            QueueUrl=queue_url,
+            AttributeNames=["QueueArn"],
+        )
+        assert "MessageRetentionPeriod" in unsupported_attributes["Attributes"].keys()
+        assert "604800" == unsupported_attributes["Attributes"]["MessageRetentionPeriod"]
+        assert "QueueArn" in supported_attributes["Attributes"].keys()
+        assert "RedrivePolicy" in unsupported_attributes["Attributes"].keys()
```

Comment:
I think we could re-write this so each line asserts the actual values, not just whether the keys in the list attributes

Reply:
I am not happy with this test in general yet, but I committed the WIP state anyway. And yes, agreed.